pages/docs/reference/data-classes.md

```diff
@@ -119,6 +119,28 @@ val olderJack = jack.copy(age = 2)
 
 </div>
 
+Note that the `copy()` call performs shallow copy, meaning that it doesn't propagate recursively through attributes:
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+data class User(var name : String)
+
+data class Address(var user: User, var city: String)
+
+var userJack = User(name="jack")
+var address = Address(user = userJack, city = "London")
+var addressCopy = address.copy()
+
+address.city = "New York"
+address.user.name = "john"
+
+println(addressCopy.city)  // prints "London"
+println(addressCopy.user.name)  // prints "john", because `address.user === addressCopy.user`
```

Comment:
We can make this same runnable for a better demonstration. I also changed the example a bit, please check that it's OK with you.
```suggestion
<div class="sample" markdown="1" theme="idea">

```kotlin
fun main() {
//sampleStart
    var userJack = User(name = "Jack")
    var address = Address(user = userJack, city = "London")
    var addressCopy = address.copy()

    address.city = "New York" 
    address.user.name = "John" // Propagates to addressCopy because they both point to userJack.
//sampleEnd
    println("addressCopy.city is ${addressCopy.city}")  // Prints "London"
    println("addressCopy.user.name is ${addressCopy.user.name}")  // Prints "John"
}
```

Reply:
Don't you need the class definitions?
```
data class User(var name : String)
data class Address(var user: User, var city: String)
```

Also I was thinking of another way to rewrite the example a bit.

Reply:
I added your suggestions in a5753bd and I also changed the code so that the copy is modified, which seems to me to be the more common mistake make by developers. Please review it.

P.S. Sorry for the long absence, I was away during the holidays. Happy New Year :)

Reply:
Seems fine now, thank you! 
Happy New Year to you too =)